src/Relation/BelongsTo.js

```diff
@@ -16,7 +16,8 @@ export const BelongsToRelation = Relation.extend({
       return
     }
     const relatedId = utils.get(record, this.foreignKey)
-    if (relatedId !== undefined) {
+
+    if (!utils.isUorN(relatedId)) {
```

Comment:
To simplify, I would just do `if (relatedId !== undefined && relatedId !== null) {`